source/slang/slang-ir-inst-defs.h

```diff
@@ -403,6 +403,7 @@ INST(HighLevelDeclDecoration,               highLevelDecl,          1, 0)
     INST(PatchConstantFuncDecoration,       patchConstantFunc,      1, 0)
 
         /// An `[entryPoint]` decoration marks a function that represents a shader entry point.
+        /// Also used in some scenarios mark parameters that are moved from entry point parameters to global params as coming from the entry point.
```

Comment:
I might argue in favor of having a distinct decoration like `[entryPointParam]`. I worry that we might have places in the code that would assume anything marked with `[entryPoint]` is, you know... an entry point. :)